src/main/java/com/synopsys/integration/jenkins/scan/service/ScannerArgumentService.java

```diff
@@ -52,17 +56,45 @@ public void setInputJsonFilePath(String inputJsonFilePath) {
-        Object scanObject = scanStrategy.prepareScanInputForBridge(scanParameters);
+    public List<String> getCommandLineArgs(Map<String, Object> scanParameters, FilePath bridgeInstallationPath) throws ScannerJenkinsException {
+        List<String> commandLineArgs = new ArrayList<>();
+        commandLineArgs.add(getBridgeRunCommand(bridgeInstallationPath));
+
+        ParametersService parametersService = new ParametersService(listener);
+        Set<String> scanTypes = parametersService.getScanTypes(scanParameters);
 
-        boolean fixPrOrPrComment = isFixPrOrPrCommentValueSet(scanObject);
+        boolean fixPrOrPrComment = isFixPrOrPrCommentValueSet(scanParameters);
 
         SCMRepositoryService scmRepositoryService = new SCMRepositoryService(listener, envVars);
         Object scmObject =  scmRepositoryService.fetchSCMRepositoryDetails(scanParameters, fixPrOrPrComment);
-
-        List<String> commandLineArgs = new ArrayList<>(getInitialBridgeArgs(scanType, bridgeInstallationPath));
-        commandLineArgs.add(createBridgeInputJson(scanObject, scmObject, fixPrOrPrComment));
+        
+        if (scanTypes.contains(ScanType.BLACKDUCK.name())) {
+            BlackDuckParametersService blackDuckParametersService = new BlackDuckParametersService(listener);
+            BlackDuck blackDuck = blackDuckParametersService.prepareBlackDuckObjectForBridge(scanParameters);
+
+            commandLineArgs.add(BridgeParams.STAGE_OPTION);
+            commandLineArgs.add(BridgeParams.BLACKDUCK_STAGE);
+            commandLineArgs.add(BridgeParams.INPUT_OPTION);
+            commandLineArgs.add(createBridgeInputJson(blackDuck, scmObject, fixPrOrPrComment, ApplicationConstants.BLACKDUCK_INPUT_PREFIX));
+        }
+        if (scanTypes.contains(ScanType.COVERITY.name())) {
+            CoverityParametersService coverityParametersService = new CoverityParametersService(listener);
+            Coverity coverity = coverityParametersService.prepareCoverityObjectForBridge(scanParameters);
+
+            commandLineArgs.add(BridgeParams.STAGE_OPTION);
+            commandLineArgs.add(BridgeParams.COVERITY_STAGE);
+            commandLineArgs.add(BridgeParams.INPUT_OPTION);
+            commandLineArgs.add(createBridgeInputJson(coverity, scmObject, fixPrOrPrComment, ApplicationConstants.COVERITY_INPUT_PREFIX));
+        }
+        if (scanTypes.contains(ScanType.POLARIS.name())) {
+            PolarisParametersService polarisParametersService = new PolarisParametersService(listener);
+            Polaris polaris = polarisParametersService.preparePolarisObjectForBridge(scanParameters);
+
+            commandLineArgs.add(BridgeParams.STAGE_OPTION);
+            commandLineArgs.add(BridgeParams.POLARIS_STAGE);
+            commandLineArgs.add(BridgeParams.INPUT_OPTION);
+            commandLineArgs.add(createBridgeInputJson(polaris, scmObject, fixPrOrPrComment, ApplicationConstants.POLARIS_INPUT_PREFIX));
+        }
```

Comment:
The entire if segment could in a separate method named: `prepareScanTypeSpecificCommands(....)`